### PR TITLE
feat: Add recipe functions to get OAuth2Clients

### PR DESCRIPTION
**Pull request by anku255**
## Summary of change

This PR adds a recipe function to get OAuth2Clients.

## Related issues

-   Link to issue1 here
-   Link to issue1 here

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work. Bonus points for screenshots and videos!)

## Documentation changes

(If relevant, please create a PR in our [docs repo](https://github.com/supertokens/docs), or create a checklist here highlighting the necessary changes)

## Checklist for important updates

-   [ ] Changelog has been updated
-   [ ] `coreDriverInterfaceSupported.json` file has been updated (if needed)
    -   Along with the associated array in `lib/ts/version.ts`
-   [ ] `frontendDriverInterfaceSupported.json` file has been updated (if needed)
-   [ ] Changes to the version if needed
    -   In `package.json`
    -   In `package-lock.json`
    -   In `lib/ts/version.ts`
-   [ ] Had run `npm run build-pretty`
-   [ ] Had installed and ran the pre-commit hook
-   [ ] If new thirdparty provider is added,
    -   [ ] update switch statement in `recipe/thirdparty/providers/configUtils.ts` file, `createProvider` function.
    -   [ ] add an icon on the user management dashboard.
-   [ ] Issue this PR against the latest non released version branch.
    -   To know which one it is, run find the latest released tag (`git tag`) in the format `vX.Y.Z`, and then find the latest branch (`git branch --all`) whose `X.Y` is greater than the latest released tag.
    -   If no such branch exists, then create one from the latest released branch.
-   [ ] If have added a new web framework, update the `add-ts-no-check.js` file to include that
-   [ ] If added a new recipe / api interface, then make sure that the implementation of it uses NON arrow functions only (like `someFunc: function () {..}`).
-   [ ] If added a new recipe, then make sure to expose it inside the recipe folder present in the root of this repo. We also need to expose its types.

### CODE REVIEW

File: lib/ts/recipe/oauth2/types.ts [hunk 1]
```diff
@@ -48,6 +48,22 @@ export type APIOptions = {
...
 
 export type RecipeInterface = {
+    getOAuth2Clients(
+        input: GetOAuth2ClientsInput,
+        userContext: UserContext
+    ): Promise<
+        | {
+              status: "OK";
+              clients: Array<OAuth2Client>;
+              nextPageToken?: string;
```
Comment (porcellus) [inline lib/ts/recipe/oauth2/types.ts:58]: Please rename this to `paginationToken` token both in the input and the return type to match `getUsersNewestFirst`

File: lib/ts/recipe/oauth2/recipeImplementation.ts [hunk 2]
```diff
@@ -30,6 +30,37 @@ export default function getRecipeInterface(
...
+                new NormalisedURLPath(`/recipe/oauth2/admin/clients`),
+                transformObjectKeys(input, "snake-case"),
+                userContext
+            );
+
+            if (response.status === "OK") {
+                let nextPageToken: string | undefined;
+                const linkHeader = response.headers.get("link");
+
+                const nextLinkMatch = linkHeader.match(/<([^>]+)>;\s*rel="next"/);
```
Comment (porcellus) [inline lib/ts/recipe/oauth2/recipeImplementation.ts:44]: Please add a short explanation comment here. An example of the header contents would be nice.

**Review by porcellus — CHANGES_REQUESTED** [review]

File: lib/ts/recipe/oauth2/types.ts (hunk 1 above)
Comment (anku255) [reply lib/ts/recipe/oauth2/types.ts]: done

File: lib/ts/recipe/oauth2/recipeImplementation.ts (hunk 2 above)
Comment (anku255) [reply lib/ts/recipe/oauth2/recipeImplementation.ts:48]: done

**Review by porcellus — APPROVED** [review]

**Pull request merged**